src/rgw/rgw_rados.cc

```diff
@@ -8741,6 +8741,8 @@ int RGWRados::get_obj_state_impl(RGWObjectCtx *rctx, const RGWBucketInfo& bucket
       s->manifest.set_head(bucket_info.placement_rule, obj, s->size); /* patch manifest to reflect the head we just read, some manifests might be
                                              broken due to old bugs */
       s->size = s->manifest.get_obj_size();
+      if (s->accounted_size < s->size)
```

Comment:
thanks for adding the `compressed` flag - can you update this check to say `if (!compressed)`?